### PR TITLE
Add meta tag data for a given page to the data object

### PR DESCRIPTION
**Pull request by nbrl**
As it says. During data processing, add the meta tags for a given page to the data object.

### CODE REVIEW

File: src/sdk.js [hunk 1]
```diff
@@ -364,10 +365,54 @@
...
+        var meta = document.getElementsByTagName('meta');
+        if (!meta || meta.length === 0) {
+            return {};
+        }
+
+        var data = {};
+
+        for (var i=0, len=meta.length; i<len; i++) {
+            var name = meta[i].name || meta[i].getAttribute('property');
+            if (name && name != 'null') {
```
Comment (jabclab) [inline src/sdk.js]: Do you want to use `!==` here?

**Comment by jabclab** [comment]
Run on local iOS simulator.

File: src/sdk.js (hunk 1 above)
Comment (jabclab) [inline src/sdk.js]: Do we need the `name !=  'null'` check?

File: src/sdk.js [hunk 2]
```diff
@@ -364,10 +365,54 @@
...
+                data[name] = meta[i].content;
```
Comment (jabclab) [inline src/sdk.js]: Verify this can never return !string

File: src/sdk.js [hunk 3]
```diff
@@ -364,10 +365,54 @@
             }
         }
 
+        // Add page metadata to data object
+        var metaTags = Amigo._scrapeMetaTags();
+        if (!isEmptyObject(metaTags)) {
```
Comment (jabclab) [inline src/sdk.js]: Let's do this for client metadata too.

Comment (nbrl) [inline src/sdk.js]: Done

File: src/sdk.js (hunk 1 above)
Comment (nbrl) [inline src/sdk.js]: Apparently not.

File: src/sdk.js (hunk 2 above)
Comment (nbrl) [inline src/sdk.js]: As expected, it's always a string, but there is now a test for it.

**Pull request merged**